--- src/quit.cpp
+++ src/quit.cpp
@@ -4,19 +4,18 @@
 
 #include <iostream>
 #include <string>
-#include <cstring>  // sigabbrev_np()
 
 
 static volatile bool quit;
 
 void signal_handler(const int signum) {
     if(quit) {
-        info("Received signal 'SIG" + STR(sigabbrev_np(signum)) + "' while quitting; will now force quit");
+        info("Received signal number " + STR(signum) + " while quitting; will now force quit");
         exit(-2);
     }
 
     std::cout << std::endl;
-    info("Signal 'SIG" + STR(sigabbrev_np(signum)) + "' received; quitting application on next frame...");
+    info("Signal number " + STR(signum) + " received; quitting application on next frame...");
 
     set_quit();
 }
